lib_xua/src/core/endpoint0/xua_ep0_descriptors.h

```diff
@@ -802,6 +812,10 @@ typedef struct
 #endif
 #endif
 
+#if XUA_USER_DESCRIPTORS_ENABLE
```

Comment:
you should be able to use #ifdef __xua_additional_descriptors_decl_exists__ here (when added to lib_build_info.cmake as an optional header. I would stick the xua_user_ naming convention here also e.g. xua_user_descriptors_ (and its shorter..)